### DISCUSSION TITLE
Wireguard vs OpenVpn

### DISCUSSION THREAD

**Discussion by davixdedem** [davixdedem/Pi-Starlink#2] [Q&A] [answered]
Hey folks, would you prefer to use **WireGuard** or **OpenVPN** to connect to your Pi Starlink from outside?

We're currently using OpenVPN because OpenVPN for Android offers straightforward APIs. Let me know what you think!

**Comment by ScriptGeek** [comment]
I think OpenVPN is the better choice if only one can be supported.

**Reply by davixdedem** [reply]
Actually we could integrate both of them. 
I read many comments under the post I made on [Reddit](https://www.reddit.com/r/Starlink/comments/1foy8ow/pi_starlink_the_raspberry_pis_unofficial_solution/) where people said that Wireguard is stronger than OpenVPN. 

Maybe It's worth trying it out.

**Reply by davixdedem** [reply]
I gave WireGuard a try, and I must say it’s noticeably faster than OpenVPN. I’m definitely considering implementing it in the next few days!

**Reply by ScriptGeek** [reply, marked as answer]
I read that OpenVPN is more widely supported and has a lot more features. It also uses TCP and UDP while WireGuard only uses UDP. It would be great to have both technologies implemented.
